tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/LiferayServerOutputStreamMonitor.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-2014 Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+package com.liferay.ide.server.core.portal;
+
+import com.liferay.ide.server.core.LiferayServerCore;
+
+import org.eclipse.core.runtime.ISafeRunnable;
+import org.eclipse.core.runtime.SafeRunner;
+import org.eclipse.debug.core.IStreamListener;
+import org.eclipse.debug.core.model.IFlushableStreamMonitor;
+
+public class LiferayServerOutputStreamMonitor implements IFlushableStreamMonitor
+{
+
+    class StreamNotifier implements ISafeRunnable
+    {
+
+        private IStreamListener _listener;
+        private String _text;
+
+        public void handleException( Throwable exception )
+        {
+            LiferayServerCore.logError( exception );
+        }
+
+        public void notifyAppend( String text )
+        {
+            if( text == null )
+            {
+                return;
+            }
+            this._text = text;
+            Object[] listeners = _listeners.getListeners();
+
+            for( int i = 0; i < listeners.length; ++i )
+            {
+                this._listener = ( (IStreamListener) listeners[i] );
+                SafeRunner.run( this );
+            }
+            this._listener = null;
+            this._text = null;
+        }
+
+        public void run() throws Exception
+        {
+            this._listener.streamAppended( this._text, LiferayServerOutputStreamMonitor.this );
+        }
+    }
+
+    PortalListenerList _listeners = new PortalListenerList( 1 );
+    private boolean _buffered = true;
+
+    private StringBuffer _contentsBuffer;
+
+    public LiferayServerOutputStreamMonitor()
+    {
+        this._contentsBuffer = new StringBuffer();
+    }
+
+    public void addListener( IStreamListener listener )
+    {
+        this._listeners.add( listener );
```

Comment:
We don't need both the `this` and the `_`.  Just use the `_` and not the `this` since that is the new Liferay standard we are going to be switching our IDE codebase to.